platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/builder/PlatformComponentBuilder.java

```diff
@@ -220,4 +224,43 @@ public EventDeduplicator buildEventDeduplicator() {
         }
         return eventDeduplicator;
     }
+
+    /**
+     * Provide an event signature validator in place of the platform's default event signature validator.
+     *
+     * @param eventSignatureValidator the event signature validator to use
+     * @return this builder
+     */
+    @NonNull
+    public PlatformComponentBuilder withEventSignatureValidator(
```

Comment:
A few unused methods in this class, old and new. If they're really unused (vs indirectly via, say, reflection), any reason for adding them?

Reply:
Although I agree that in general it's important to remove unused methods, these particular methods are intentionally not removed. The purpose of this class is to allow us to swap out component implementations at platform construction time. We currently don't swap anything out, but we will do so in the near future.